Mathlib/Algebra/Group/Submonoid/Basic.lean

```diff
@@ -572,8 +588,8 @@ def ofClosureMEqTopLeft {M N} [Monoid M] [Monoid N] {s : Set M} (f : M → N) (h
   toFun := f
   map_one' := h1
   map_mul' x :=
-    (dense_induction (p := _) x hs hmul fun y => by rw [one_mul, h1, one_mul]) fun a b ha hb y => by
-      rw [mul_assoc, ha, ha, hb, mul_assoc]
+    dense_induction (p := _) _ hs hmul fun y => by rw [one_mul, h1, one_mul]
```

Comment:
```suggestion
    dense_induction _ hs hmul fun y => by rw [one_mul, h1, one_mul]
```
untested, but this looks like the sort of explicit insertion that isn't really doing anything?

Reply:
I thought the same thing. It's necessary, but I'm not sure why. Note that it was already there.

Reply:
Maybe it forces the elaboration order to change, which turns out to be necessary here